tests/generators/kzg_4844/main.py

```diff
@@ -0,0 +1,580 @@
+"""
+KZG 4844 test vectors generator
+"""
+
+from hashlib import sha256
+from typing import Tuple, Iterable, Any, Callable, Dict
+
+from eth_utils import (
+    encode_hex,
+    int_to_big_endian,
+)
+
+from eth2spec.utils import bls
+from eth2spec.test.helpers.constants import DENEB
+from eth2spec.test.helpers.typing import SpecForkName
+from eth2spec.gen_helpers.gen_base import gen_runner, gen_typing
+from eth2spec.deneb import spec
+
+
+def expect_exception(func, *args):
+    try:
+        func(*args)
+    except Exception:
+        pass
+    else:
+        raise Exception("should have raised exception")
+
+
+def field_element_bytes(x):
+    return int.to_bytes(x % spec.BLS_MODULUS, 32, "little")
+
+
+def encode_hex_list(a):
+    return [encode_hex(x) for x in a]
+
+
+def bls_add_one(x):
+    """
+    Adds "one" (actually bls.G1()) to a compressed group element.
+    Useful to compute definitely incorrect proofs.
+    """
+    return bls.G1_to_bytes48(
+        bls.add(bls.bytes48_to_G1(x), bls.G1())
+    )
+
+
+def evaluate_blob_at(blob, z):
+    return field_element_bytes(
+        spec.evaluate_polynomial_in_evaluation_form(spec.blob_to_polynomial(blob), spec.bytes_to_bls_field(z))
+    )
+
+
+P1_NOT_IN_G1 = bytes.fromhex("8123456789abcdef0123456789abcdef0123456789abcdef" +
+                             "0123456789abcdef0123456789abcdef0123456789abcdef")
+P1_NOT_ON_CURVE = bytes.fromhex("8123456789abcdef0123456789abcdef0123456789abcdef" +
+                                "0123456789abcdef0123456789abcdef0123456789abcde0")
+BLS_MODULUS_BYTES = spec.BLS_MODULUS.to_bytes(32, spec.ENDIANNESS)
+
+BLOB_ALL_ZEROS = spec.Blob()
+BLOB_RANDOM_VALID1 = spec.Blob(b''.join([field_element_bytes(pow(2, n + 256, spec.BLS_MODULUS)) for n in range(4096)]))
+BLOB_RANDOM_VALID2 = spec.Blob(b''.join([field_element_bytes(pow(3, n + 256, spec.BLS_MODULUS)) for n in range(4096)]))
+BLOB_RANDOM_VALID3 = spec.Blob(b''.join([field_element_bytes(pow(5, n + 256, spec.BLS_MODULUS)) for n in range(4096)]))
+BLOB_ALL_MODULUS_MINUS_ONE = spec.Blob(b''.join([field_element_bytes(spec.BLS_MODULUS - 1) for n in range(4096)]))
+BLOB_ALMOST_ZERO = spec.Blob(b''.join([field_element_bytes(1 if n == 3211 else 0) for n in range(4096)]))
+BLOB_INVALID = spec.Blob(b'\xFF' * 4096 * 32)
+BLOB_INVALID_CLOSE = spec.Blob(b''.join(
+    [BLS_MODULUS_BYTES if n == 2111 else field_element_bytes(0) for n in range(4096)]
+))
+
+VALID_BLOBS = [BLOB_ALL_ZEROS, BLOB_RANDOM_VALID1, BLOB_RANDOM_VALID2,
+               BLOB_RANDOM_VALID3, BLOB_ALL_MODULUS_MINUS_ONE, BLOB_ALMOST_ZERO]
+INVALID_BLOBS = [BLOB_INVALID, BLOB_INVALID_CLOSE]
+VALID_ZS = [field_element_bytes(x) for x in [0, 1, 2, pow(5, 1235, spec.BLS_MODULUS), spec.BLS_MODULUS - 1]]
```

Comment:
Should we maybe add a non-1 root of unity here as well, for a deeper check of the in-domain compute ZKG proof and evaluation formulas?

Reply:
`spec.BLS_MODULUS - 1` is such a non-` root of unity, but sure I'll add another one.